core/capabilities/streams/trigger_load_test.go

```diff
@@ -0,0 +1,224 @@
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote"
+	remotetypes "github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/streams"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/chaintype"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ocr2key"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+const (
+	triggerID          = "streams-trigger@1.0.0"
+	workflowID         = "workflowID1"
+	baseTriggerEventID = "triggerEvent_"
+	basePrice          = 2000000000
+	baseTimestamp      = 1000000000
+)
+
+type feed struct {
+	feedID    [32]byte
+	feedIDStr string
+	reports   []report
+}
+
+type report struct {
+	rawReport  []byte
+	reportCtx  []byte
+	signatures [][]byte
+}
+
+type node struct {
+	peerID p2ptypes.PeerID
+	bundle ocr2key.KeyBundle
+}
+
+// Load test that combines Trigger Subscriber, Streams Trigger Aggregator and Streams Codec.
+// It measures time needed to receive and process trigger events from multiple nodes and produce a local aggregated event.
+// For more meaningful measurements, increase the values of parameters P and T.
+func TestStreamsTrigger_Load(t *testing.T) {
+	N := 31 // trigger DON nodes
+	F := 10 // faulty nodes
+	R := 5  // different reports per feed (i.e. prices and timestamps)
+	P := 2  // feeds
+	T := 2  // test iterations
+
+	nodes := newNodes(t, N)
+	feeds := newFeedsWithSignedReports(t, nodes, N, P, R)
+
+	allowedSigners := make([][]byte, N)
+	for i := 0; i < N; i++ {
+		allowedSigners[i] = nodes[i].bundle.PublicKey() // bad name - see comment on evmKeyring.PublicKey
+	}
+	lggr := logger.TestLogger(t)
+	ctx := testutils.Context(t)
+	codec := streams.NewCodec(lggr)
+	agg := triggers.NewMercuryRemoteAggregator(codec, allowedSigners, F, lggr)
+
+	capInfo := capabilities.CapabilityInfo{
+		ID: triggerID,
+	}
+	capMembers := make([]p2ptypes.PeerID, N)
+	for i := 0; i < N; i++ {
+		capMembers[i] = nodes[i].peerID
+	}
+	capDonInfo := capabilities.DON{
+		Members: capMembers,
+		F:       uint8(F),
+	}
+	config := &remotetypes.RemoteTriggerConfig{
+		MinResponsesToAggregate: uint32(F + 1),
+	}
+	subscriber := remote.NewTriggerSubscriber(config, capInfo, capDonInfo, capabilities.DON{}, nil, agg, lggr)
+
+	// register trigger
+	req := capabilities.CapabilityRequest{
+		Metadata: capabilities.RequestMetadata{
+			WorkflowID: workflowID,
+		},
+	}
+	triggerEventCallbackCh, err := subscriber.RegisterTrigger(ctx, req)
+	require.NoError(t, err)
+
+	// send and process all trigger events
+	startTs := time.Now().UnixMilli()
+	processingTime := int64(0)
+	for c := 0; c < T; c++ {
```

Comment:
does this log any status or progress? i image this test needs to be run with some sort of timeout. if it runs for 5 minutes and the test ctx is cancelled, will you know how far this got relative to the total work to be done?

Reply:
So far I've only relied on the log from line 137 below. I doubt there will ever be a need to run it for periods longer than test framework timeouts. If test fails for any reason then we won't get the log summary (or hit an assert) and that won't produce meaningful results.